DcNotificationService/NotificationService.swift

```diff
@@ -14,21 +14,36 @@ class NotificationService: UNNotificationServiceExtension {
             contentHandler(silenceNotification())
             return
         }
-        UserDefaults.setNseFetching()
+        UserDefaults.setNseFetching(until: Date().addingTimeInterval(26))
 
         // as we're mixing in notifications from accounts without PUSH and we cannot add multiple notifications,
         // it is best to move everything to the same thread - and set just no threadIdentifier
 
         dcAccounts.openDatabase(writeable: false)
         let eventEmitter = dcAccounts.getEventEmitter()
 
-        if !dcAccounts.backgroundFetch(timeout: 25) {
-            UserDefaults.pushToDebugArray("ERR3")
-            UserDefaults.setNseFetching(false)
+        // Send the bestAttempt notification when memory is critical because the process will be killed
+        // by the system soon and any notification is better than nothing.
+        var exitedDueToCriticalMemory = false
+        let memoryPressureSource = DispatchSource.makeMemoryPressureSource(eventMask: .critical)
+        memoryPressureSource.setEventHandler {
+            // Order of importance because we might crash very soon
+            // TODO: Could notify with "received message too big to decrypt in background"
             contentHandler(bestAttemptContent)
+            exitedDueToCriticalMemory = true
+            UserDefaults.pushToDebugArray("ERR5")
+        }
+        memoryPressureSource.activate()
+
+        guard dcAccounts.backgroundFetch(timeout: 25) && !exitedDueToCriticalMemory else {
```

Comment:
i find that hard to read, also not sure if it is correct (if backgroundFetch succeeds, exitedDueToCriticalMemory is ignored)

i would expect the following:

```suggestion
        if !dcAccounts.backgroundFetch(timeout: 25) || exitedDueToCriticalMemory {
```

Reply:
guard implies it does not continue unless the values are true, which I think makes it more readable. Previously there was an if statement with a return, which is A hard to read and B is not compile safe (the return can be deleted in the future)

the guard basically means "if bgfetch returns true and not exitedDueToCriticalMemory, continue, else return in this block"

so exitedDueToCriticalMemory is not ignored.